sklearn/preprocessing/_encoders.py

```diff
@@ -161,6 +161,19 @@ class OneHotEncoder(_BaseEncoder):
 
         The used categories can be found in the ``categories_`` attribute.
 
+    drop : 'first' or a list/array of shape (n_features,), default=None.
+        Specifies a methodology to use to drop one of the categories per
+        feature. This is useful in situations where perfectly collinear
+        features cause problems, such as when feeding the resulting data
+        into a neural network or an unregularized regression. If only one
+        feature appears in a column, that column will be retained as a
+        column of ones.
```

Comment:
Why should it be retained?

Reply:
No real preference here. I've changed it so it will be dropped instead.